sdk/common/src/main/java/io/opentelemetry/sdk/common/CompletableResultCode.java

```diff
@@ -123,9 +123,10 @@ public CompletableResultCode fail() {
    * Completes this {@link CompletableResultCode} unsuccessfully if it is not already completed,
    * setting the {@link #getFailureThrowable() failure throwable} to {@code throwable}.
    *
+   * @param throwable the {@code Throwable} that caused the failure, or {@code null}
    * @since 1.41.0
    */
-  public CompletableResultCode failExceptionally(Throwable throwable) {
+  public CompletableResultCode failExceptionally(@Nullable Throwable throwable) {
```

Comment:
#nit: do you think we should update the javadoc for [getFailureThrowable](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/common/src/main/java/io/opentelemetry/sdk/common/CompletableResultCode.java#L157-L165) to indicate this new case for a null result? I.e. 
```
@return the throwable if failed exceptionally, or null if: {@link #fail() failed without
   *     exception}, {@link #succeed() succeeded}, {@link
   * #failExceptionally(Throwable)} with a null {@code throwable}, or not complete.
```

Reply:
yeah, seems good. I'll add it.